src/black/trans.py

```diff
@@ -590,11 +590,21 @@ def make_naked(string: str, string_prefix: str) -> str:
             """
             assert_is_leaf_string(string)
             if "f" in string_prefix:
-                string = _toggle_fexpr_quotes(string, QUOTE)
-                # After quotes toggling, quotes in expressions won't be escaped
-                # because quotes can't be reused in f-strings. So we can simply
-                # let the escaping logic below run without knowing f-string
-                # expressions.
+                f_expressions = (
+                    string[span[0] : span[1]] for span in iter_fexpr_spans(string)
+                )
+                expressions_contain_printable_quotes = any(
+                    re.search(r".*[\'\"].*=", expression)
```

Comment:
Something like this could avoid false triggering on `f"{x == y}"`
```suggestion
                    re.fullmatch(r".*[\'\"].*=\s*", expression)
```

Reply:
`f"{foo["bar"]=:.2f"` would still be a problem. Maybe `re.fullmatch(r"{.*[\'\"][^=]*=[^=]*}", expression)`?

Reply:
Oh good point. I think that won't work with `!=` and `:=` and so on. I think maybe let's just keep this as in the current version of the PR and revisit when PEP 701 support is landed